siege/siege.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
+#
+# Compares throughput from two different versions of Envoy, with multiple
+# interleaved iterations. The two binaries are supplied on the commmand-line:
+#
+# Usage:
+#    siege.py clean_envoy_binary experimental_envoy_binary output_dir
+#
+# The two binaries should generally be built with
+#     bazel build -c opt source/exe:envoy-static
+# and can then be copied to a safe place (e.g. /tmp) so that they can each
+# be based on a single git workspace that you flip between branches.
+#
+# output_dir is where the logs and raw CSV files get written, but the
+# the summarized performance data is printed as a table to stdout.
+
+import json
+import os
+import requests
+import subprocess
+import sys
+import time
+import urllib.request
+
+# TODO(jmarantz): make the configuration pluggable, which means grepping
+# for these ports rather than hardcoding them. It's annoying to grep in
+# yaml files because there isn't per-line context. Another possibility
+# is to scrape them from Envoy log output.
+proxy_port = 10001
+upstream_port = 10000
+admin_port = 8081
+
+upstream_url = "http://127.0.0.1:%d" % upstream_port
+proxy_url = "http://127.0.0.1:%d" % proxy_port
+
+# We use the admin port both for scraping memory and for quitting at
+# the end of each run.
+admin = "http://127.0.0.1:%d" % admin_port
+
+def main(argv):
+  if len(argv) != 4:
+    print("Usage: %s clean_envoy_binary experimental_envoy_binary output_dir"
+          % argv[0])
+    sys.exit(1)
+
+  # Capture arguments as locals for readability.
+  clean_envoy = argv[1]
```

Comment:
I think that with the number of possible configuration options, e.g. ports, debug/quiet modes as well as positional args, that `argparse` would make sense (https://docs.python.org/3/library/argparse.html).